include/xsimd/config/xsimd_config.hpp

```diff
@@ -238,6 +238,10 @@
 #define XSIMD_WITH_SSE2 1
 #endif
 
+#if defined(_M_X64) || (defined(_M_IX86_FP) && _M_IX86_FP >= 2)
+#define XSIMD_WITH_SSE2 1
+#endif
```

Comment:
If we only enable SSE2 here, then SSE3,SSE4 will not be supported by xsimd if built with msvc.
I think this is a problem and will still break many code.
MSVC X64 must support SSE4_2 (actually /arch:SSE2 is an invalid option for msvc x64).
What about define `XSIMD_WITH_SSE4_2` if _M_X64?

Reply:
According to the doc there's indeed no explicit flag for SSE3 and later (based on https://docs.microsoft.com/fr-fr/cpp/build/reference/arch-x86?view=msvc-160) but I wonder if it's actually supported if no arch flag is specified ?

Reply:
X64 msvc only supports /arch:AVX?, https://docs.microsoft.com/en-us/cpp/build/reference/arch-x64?view=msvc-160

My understanding is /arch:XXX anables msvc to auto vectorize c++ code using at most the XXX instructions.
But it doesn't prevent us from using intrinsics directly. Compiler should supports all level simd instructions.